### DISCUSSION TITLE
Replace legacy MSBuild brand with some fancy vector graphic

### DISCUSSION THREAD

**Discussion by deeprobin** [dotnet/msbuild#7712] [Ideas] [answered]
The logo which was initiated by @terrajobst to my knowledge once is not bad, but the resolution and the design is in my opinion no longer up to date.

What do you think about creating a more modern logo? Personally, I am not a graphic designer, but I would like to see a logo in a modern flat style (maybe in a color similar to the .NET Foundation logo).

**Comment by terrajobst** [comment]
I built the larger 256x256 version of the icon in Photoshop by overlaying the low res version of the icon embedded in msbuild.exe. In principle one could redo this with, say, Inksape, as an SVG as well.

Where is the image being featured, by the way?

**Comment by teo-tsirpanis** [comment]
> Where is the image being featured, by the way?

One place the logo is featured is in the Visual Studio distribution of MSBuild.

![image](https://user-images.githubusercontent.com/12659251/173935174-55a6b9e7-370c-4c11-be01-65fd752ef36a.png)

Making it an SVG would be a nice addition and enable displaying the icon in more resolutions.

> the design is in my opinion no longer up to date

It might be inconsitent with other .NET/VS icons but I personally find it quite good on its own.

**Comment by terrajobst** [comment]
> One place the logo is featured is in the Visual Studio distribution of MSBuild.

Not quite. I based my version off of the low res version contained in msbuild.exe; the embedded icon maxes out at 32x32. However, someone could add my 256x256 version (and maybe even a downscaled 128x, 64x, and 48x, for good measure).

> Making it an SVG would be a nice addition and enable displaying the icon in more resolutions.

You mean in this repo? File icons can't be SVGs AFAIK.

> It might be inconsitent with other .NET/VS icons but I personally find it quite good on its own.

Agreed.

**Comment by teo-tsirpanis** [comment]
> File icons can't be SVGs AFAIK.

Yes but it would make creating a multi-res `.ico` file more easy.

**Comment by drewnoakes** [comment, marked as answer]
The icon was updated since this question was asked.

https://github.com/dotnet/msbuild/tree/main/branding

![](https://github.com/dotnet/msbuild/blob/main/branding/MSBuild-Logo.svg?raw=true)
